src/mongo/mongoConnectionStrings.ts

```diff
@@ -13,7 +13,7 @@
 //   Full match
 //   mongodb[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]]
 //   [database]
-const parsePrefix = '^(mongodb(?:[+]srv)?:\/\/[^\/]*)';
+const parsePrefix = '([a-zA-Z]+:\/\/[^\/]*)';
```

Comment:
You removed the ^ and $.  But this is better anyway (^ and $ shouldn't be in the component regexes - separation of concerns).

const parseConnectionString = new RegExp(`^${parsePrefix}${parseDatabaseName}\$`);

Reply:
… which indicates you should add some tests that have extra characters at the beginning and end of the string

Reply:
Never mind... :-)